sql/src/test/java/org/apache/druid/sql/calcite/PlannerContextTest.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.druid.sql.calcite.planner.CatalogResolver;
+import org.apache.druid.sql.calcite.planner.NoOpPlannerHook;
+import org.apache.druid.sql.calcite.planner.PlannerConfig;
+import org.apache.druid.sql.calcite.planner.PlannerContext;
+import org.apache.druid.sql.calcite.planner.PlannerToolbox;
+import org.apache.druid.sql.calcite.run.NativeSqlEngine;
+import org.apache.druid.sql.calcite.schema.DruidSchema;
+import org.apache.druid.sql.calcite.schema.DruidSchemaCatalog;
+import org.apache.druid.sql.calcite.schema.NamedDruidSchema;
+import org.apache.druid.sql.calcite.schema.NamedViewSchema;
+import org.apache.druid.sql.calcite.schema.ViewSchema;
+import org.apache.druid.sql.calcite.util.CalciteTests;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+public class PlannerContextTest
+{
+
+  private static final PlannerToolbox PLANNER_TOOLBOX = new PlannerToolbox(
+      CalciteTests.createOperatorTable(),
+      CalciteTests.createExprMacroTable(),
+      CalciteTests.getJsonMapper(),
+      new PlannerConfig(),
+      new DruidSchemaCatalog(
+          EasyMock.createMock(SchemaPlus.class),
+          ImmutableMap.of(
+              "druid", new NamedDruidSchema(EasyMock.createMock(DruidSchema.class), "druid"),
+              NamedViewSchema.NAME, new NamedViewSchema(EasyMock.createMock(ViewSchema.class))
+          )
+      ),
+      CalciteTests.createJoinableFactoryWrapper(),
+      CatalogResolver.NULL_RESOLVER,
+      "druid",
+      new CalciteRulesManager(ImmutableSet.of()),
+      CalciteTests.TEST_AUTHORIZER_MAPPER,
+      AuthConfig.newBuilder().build()
+  );
+
+  private static final NativeSqlEngine ENGINE = CalciteTests.createMockSqlEngine(
+      EasyMock.createMock(QuerySegmentWalker.class),
+      EasyMock.createMock(QueryRunnerFactoryConglomerate.class)
+  );
+
+  private static final String SQL = "SELECT 1";
+
+  private static PlannerContext plannerContext;
+
+  @BeforeEach
+  public void setup()
+  {
+    plannerContext = PlannerContext.create(PLANNER_TOOLBOX,
+                          SQL, // The actual query isn't important for this test
+                          ENGINE,
+                          Collections.emptyMap(),
+                          null);
+  }
+
+  @Test
+  public void testCreate()
+  {
+    Assert.assertEquals(PLANNER_TOOLBOX, plannerContext.getPlannerToolbox());
+    Assert.assertEquals(SQL, plannerContext.getSql());
+    Assert.assertEquals(ENGINE, plannerContext.getEngine());
+    Assert.assertEquals(NoOpPlannerHook.INSTANCE, plannerContext.getPlannerHook());
+    Assert.assertTrue(plannerContext.queryContextMap().isEmpty());
+  }
+
+  @Test
+  public void testMutabilityOfQueryContextMap()
+  {
+    // Validate that the map is mutable, even though an immutable map was passed during creation.
+    plannerContext.queryContextMap().put("test", "value");
```

Comment:
This test doesn't assert anything. Just verifying that put doesn't throw an exception is probably not enough.
A better verification would be to `get()` the value of the key from the map and check it against the original value used in the `put()`.